kernel/sched.c

```diff
@@ -711,9 +735,12 @@ static void z_thread_halt(struct k_thread *thread, k_spinlock_key_t key,
 						  : _THREAD_SUSPENDED);
 	}
 
-	bool active = thread_active_elsewhere(thread);
+	struct _cpu *cpu = NULL;
+	bool active = thread_active_elsewhere(thread, &cpu);
```

Comment:
As above, if you rework that predicate to be e.g. "`struct _cpu *thread_active_on(struct k_thread *);`" it would save a few lines and clean up the expression here.